src/DefinedSchemas.js

```diff
@@ -0,0 +1,308 @@
+          return;
+        }
+
+        // Check if something changed other than the type (like required, defaultValue)
+        if (!this.paramsAreEquals(field, localField)) {
+          fieldsWithChangedParams.push(fieldName);
+        }
+      });
+
+    fieldsToDelete.forEach(fieldName => {
+      newLocalSchema.deleteField(fieldName);
+    });
+
+    // Delete fields from the schema then apply changes
+    await this.updateSchemaToDB(newLocalSchema);
+
+    fieldsToRecreate.forEach(fieldName => {
+      const { type, ...others } = localSchema.fields[fieldName];
+      this.handleFields(newLocalSchema, fieldName, type, others);
+    });
+    fieldsWithChangedParams.forEach(fieldName => {
+      const { type, ...others } = localSchema.fields[fieldName];
+      this.handleFields(newLocalSchema, fieldName, type, others);
+    });
+
+    // Handle Indexes
+    // Check addition
+    if (localSchema.indexes) {
+      Object.keys(localSchema.indexes).forEach(indexName => {
+        if (
+          (!cloudSchema.indexes || !cloudSchema.indexes[indexName]) &&
+          !this.isProtectedIndex(localSchema.className, indexName)
+        )
+          newLocalSchema.addIndex(indexName, localSchema.indexes[indexName]);
+      });
+    }
+
+    const indexesToAdd = [];
+
+    // Check deletion
+    if (cloudSchema.indexes) {
+      Object.keys(cloudSchema.indexes).forEach(async indexName => {
+        if (!this.isProtectedIndex(localSchema.className, indexName)) {
+          if (!localSchema.indexes || !localSchema.indexes[indexName]) {
+            newLocalSchema.deleteIndex(indexName);
+          } else if (
+            !this.paramsAreEquals(localSchema.indexes[indexName], cloudSchema.indexes[indexName])
+          ) {
+            newLocalSchema.deleteIndex(indexName);
+            indexesToAdd.push({
+              indexName,
+              index: localSchema.indexes[indexName],
+            });
+          }
+        }
+      });
+    }
+
+    this.handleCLP(localSchema, newLocalSchema, cloudSchema);
+    // Apply changes
+    await this.updateSchemaToDB(newLocalSchema);
+    // Apply new/changed indexes
+    if (indexesToAdd.length) {
+      indexesToAdd.forEach(o => newLocalSchema.addIndex(o.indexName, o.index));
+      await this.updateSchemaToDB(newLocalSchema);
+    }
+  }
+
+  handleCLP(localSchema, newLocalSchema, cloudSchema) {
+    if (!localSchema.classLevelPermissions && !cloudSchema) {
+      logger.warn(`classLevelPermissions not provided for ${localSchema.className}.`);
+    }
+    // Use spread to avoid read only issue (encountered by Moumouls using directAccess)
+    const clp = { ...localSchema.classLevelPermissions } || {};
+    const cloudCLP = (cloudSchema && cloudSchema.classLevelPermissions) || {};
+    // Try to inject default CLPs
+    const CLPKeys = ['find', 'count', 'get', 'create', 'update', 'delete', 'addField'];
+    CLPKeys.forEach(key => {
+      if (!clp[key]) {
+        clp[key] = cloudCLP[key] || { '*': true };
```

Comment:
@Moumouls @mtrezza  Please correct me if i'm wrong, but it seems from this that when no CLP is defined in the schema,  the *default*  is Read/Write/Add  for anonymous ?

For a security perspective, i think this should have **no** permissions by default, or throw an error `Missing CLP for "ObjectName"`.

I understand the behaviour from `ParseDashboard` but from the dashboard when the user add a Class, it's created by default public, BUT there is a clear indication at the top that the class is public ...

With the schema migration behaviour being accessible by default to `anonymous` users, the developer will probably have no idea his class is public. It seems dangerous, and not a "secure by default" approach.

I think it would be resonable to either of the following : 
- Set everything to private and display a warning  `CLP is not defined for "ObjectName", the CLP has been set to PRIVATE and the objects will only be accessible using a  MASTER_KEY.`
- Throw an error until any kind of CLP is set in the schema `Missing CLP for "ObjectName"`

@++
Samuel